readme.md

```diff
@@ -48,11 +56,9 @@ There are 3 composite types in Bru.
 * [Multiline String](https://en.wikipedia.org/wiki/Here_document)
 
 #### Multimap
-Multimap is essentially a dictionary (key-value pair) that can have duplicate keys.
-Its enclosed in curly braces `{}`.
-Keys and Values are separated by a colon `:` and key-value pairs are separated by a newline `\n`.
+Multimap is a dictionary (key-value pair) with duplicate keys, enclosed in curly braces (`{}`). Keys and Values are separated by a colon `:` and key-value pairs are separated by a newline (`\n`).
 
-All keys are strings. Values can be primitive types or composite types.
+All keys are unquoted strings and must not contain spaces. Values may be primitive or composite types. Some contexts may offer further restrictions not required by the Bru language.
```

Comment:
It's likely that we want to be more specific about this. As I noted, "some contexts" are more restrictive on key names (e.g., headers *must* be legal HTTP header name values, so `[-a-zA-Z0-9]+`).

Reply:
> All keys are unquoted strings and must not contain spaces.

I think we should allow all characters in a key. Here is how I am thinking
* They key have any unicode character expect (`:`), if it has `:`, then it must be eccaped
* The key cannot be surrounded by whitespaces
* The surrounding whitespaces are trimmed while parsing.

@halostatue I think a live playground with a parser transforming `bru` to `json` would convey my motivation here.

Again, This is not set in stone. Would love to discuss nuances in a call.

Reply:
Or keys with special characters (whitespace, `:`, etc.) must be quoted. I was trying to infer some rules based on what I had seen. I would keep a variation of the `some contexts…` comment, as header keys *must* conform to a very specific ruleset.